[Bug] e2e test of raycluster_test.go failed with kindest/node:v1.24.0

### Search before asking

- [x] I searched the [issues](https://github.com/ray-project/kuberay/issues) and found no similar issues.


### KubeRay Component

ray-operator

### What happened + What you expected to happen

I follow the [ray-operator/DEVELOPMENT.md](https://github.com/ray-project/kuberay/blob/master/ray-operator/DEVELOPMENT.md) to build raycluster and run e2e test but failed in `TestRayClusterManagedBy/Failed_creation_of_cluster,_managed_by_external_controller`
I've tried run the same example in M3(arm) MacOS and x86-64 Ubuntu system. They both encountered the same error.
Later, I tried using `kindest/node:v1.26.0` instead of `kindest/node:v1.24.0` as recommended in the document and passed the test in both machines.

https://github.com/ray-project/kuberay/blob/41ee4dbd74f5d47d26d2fd0f578d1b2d1f3741fc/ray-operator/test/e2e/raycluster_test.go#L67-L76

Here's my error log
```bash
=== NAME  TestRayClusterManagedBy
    raycluster_test.go:34: Created RayCluster test-ns-26p7c/raycluster-ok successfully
    raycluster_test.go:36: Waiting for RayCluster test-ns-26p7c/raycluster-ok to become ready
    raycluster_test.go:75: 
        Expected an error to have occurred.  Got:
            <nil>: nil
    raycluster_test.go:50: Created RayCluster test-ns-26p7c/raycluster-skip successfully
    raycluster_test.go:52: RayCluster test-ns-26p7c/raycluster-skip will not become ready - not reconciled
    test.go:98: Creating ephemeral output directory as KUBERAY_TEST_OUTPUT_DIR env variable is unset
    test.go:101: Output directory has been created at: /var/folders/8n/8nvksnv52h18kybn1kfxby500000gn/T/TestRayClusterManagedBy3615644342/001
--- FAIL: TestRayClusterManagedBy (0.02s)
    --- FAIL: TestRayClusterManagedBy/Failed_creation_of_cluster,_managed_by_external_non_supported_controller (0.01s)
panic: test executed panic(nil) or runtime.Goexit
```

### Reproduction script

```bash
kind create cluster --image=kindest/node:v1.24.0
IMG=kuberay/operator:nightly make docker-build
kind load docker-image kuberay/operator:nightly
helm install kuberay-operator --set image.repository=kuberay/operator --set image.tag=nightly ../helm-chart/kuberay-operator
```
wait until kuberay-operator in `running` state
```bash
go test -timeout 30m -v ./test/e2e/raycluster_test.go ./test/e2e/support.go
```

### Anything else

_No response_

### Are you willing to submit a PR?

- [x] Yes I am willing to submit a PR!

--- a/apiserver/hack/kind-cluster-config.yaml
+++ b/apiserver/hack/kind-cluster-config.yaml
@@ -2,7 +2,7 @@ kind: Cluster
 apiVersion: kind.x-k8s.io/v1alpha4
 nodes:
 - role: control-plane
-  image: kindest/node:v1.23.17@sha256:59c989ff8a517a93127d4a536e7014d28e235fb3529d9fba91b3951d461edfdb
+  image: kindest/node:v1.25.0@sha256:428aaa17ec82ccde0131cb2d1ca6547d13cf5fdabcc0bbecf749baa935387cbf
   kubeadmConfigPatches:
     - |
       kind: InitConfiguration

--- a/ci/kind-config-buildkite.yml
+++ b/ci/kind-config-buildkite.yml
@@ -9,6 +9,7 @@ networking:
 # https://blog.scottlowe.org/2019/07/30/adding-a-name-to-kubernetes-api-server-certificate/
 nodes:
 - role: control-plane
+  image: kindest/node:v1.25.0@sha256:428aaa17ec82ccde0131cb2d1ca6547d13cf5fdabcc0bbecf749baa935387cbf
   kubeadmConfigPatches:
   - |
     kind: ClusterConfiguration

--- a/ray-operator/DEVELOPMENT.md
+++ b/ray-operator/DEVELOPMENT.md
@@ -110,7 +110,7 @@ cd ..
 
 ```bash
 # Step 1: Create a Kind cluster
-kind create cluster --image=kindest/node:v1.24.0
+kind create cluster --image=kindest/node:v1.25.0
 
 # Step 2: Install CRDs
 make -C ray-operator install

